core/src/crds_gossip_pull.rs

```diff
@@ -74,12 +75,8 @@ impl CrdsFilter {
         ((num_items / max_items).log2().ceil()).max(0.0) as u32
     }
     pub fn hash_as_u64(item: &Hash) -> u64 {
-        let arr = item.as_ref();
-        let mut accum = 0;
-        for (i, val) in arr.iter().enumerate().take(8) {
-            accum |= (u64::from(*val)) << (i * 8) as u64;
-        }
-        accum
+        let arr: [u8; 8] = item.as_ref()[..8].try_into().unwrap();
+        unsafe { std::mem::transmute::<_, u64>(arr) }.to_le()
```

Comment:
Can the byteorder crate do the same thing without unsafe here with read_u64:
`https://docs.rs/byteorder/1.3.4/byteorder/trait.ReadBytesExt.html#method.read_u64`

Reply:
turns out `unsafe` was unnecessary. we could just use `u64::from_le_bytes`. updated the commit.